src/Serilog.Sinks.Datadog.Logs/Sinks/Datadog/DatadogHttpClient.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2019 Datadog, Inc.
+
+using System;
+using System.Threading.Tasks;
+using System.Text;
+using Serilog.Debugging;
+using System.Net.Http;
+using Serilog.Events;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Serilog.Sinks.Datadog.Logs
+{
+    public class DatadogHttpClient : IDatadogClient
+    {
+        private readonly DatadogConfiguration _config;
+        private readonly string _url;
+        private const string _content = "application/json";
+        private readonly LogFormatter _formatter;
+        private HttpClient _client;
+        private readonly Encoding utf8 = Encoding.UTF8;
+        private const int _maxSize = 2 * 1024 * 1024 - 51;  // Need to reserve space for at most 49 "," and "[" + "]"
+        private const int _maxMessageSize = 256 * 1024;
+
+        /// <summary>
+        /// Max number of retries when sending failed.
+        /// </summary>
+        private const int MaxRetries = 1;
```

Comment:
Wondering whether we want to drop logs now, cc @NBParis

Reply:
Which means that we would not need a max retry.

Reply:
Two important factors here:

- We do not want to loose logs
- We do not want to impact performances of the application because logging is blocked.

Therefore we can imagine a max retry to limit the impact on the application but by default it could be something like 10 (exponential?) retries.